lib/Runtime/Library/WebAssemblyModule.cpp

```diff
@@ -281,9 +282,14 @@ WebAssemblyModule::ValidateModule(
     {
         Js::AutoDynamicCodeReference dynamicFunctionReference(scriptContext);
         SRCINFO const * srcInfo = scriptContext->cache->noContextGlobalSourceInfo;
-        Js::Utf8SourceInfo* utf8SourceInfo = Utf8SourceInfo::New(scriptContext, (LPCUTF8)buffer, lengthBytes / sizeof(char16), lengthBytes, srcInfo, false);
 
-        Wasm::WasmModuleGenerator bytecodeGen(scriptContext, utf8SourceInfo, (byte*)buffer, lengthBytes);
+        // review: unsure if we need copy here, but seems safer to do it
```

Comment:
But I don't know about if the lifetime of the UTF8SourceInfo is different. It has a finalizer and the actual ArrayBuffer buffer isn't recycler allocated, so a reference to that wouldn't keep it alive.